stdlib/cgi/0/core.rbs

```diff
@@ -560,36 +490,523 @@ class CGI
+  #     `options_hash` form, since it also allows you specify the charset you will
+  #     accept.
+  # `options_hash`
+  # :   A Hash that recognizes three options:
+  #
+  #     `:accept_charset`
+  # :       specifies encoding of received query string.  If omitted,
+  #         `@@accept_charset` is used.  If the encoding is not valid, a
+  #         CGI::InvalidEncoding will be raised.
+  #
+  #         Example. Suppose `@@accept_charset` is "UTF-8"
+  #
+  #         when not specified:
+  #
+  #             cgi=CGI.new      # @accept_charset # => "UTF-8"
+  #
+  #         when specified as "EUC-JP":
+  #
+  #             cgi=CGI.new(:accept_charset => "EUC-JP") # => "EUC-JP"
+  #
+  #     `:tag_maker`
+  # :       String that specifies which version of the HTML generation methods to
+  #         use.  If not specified, no HTML generation methods will be loaded.
+  #
+  #         The following values are supported:
+  #
+  #         "html3"
+  # :           HTML 3.x
+  #         "html4"
+  # :           HTML 4.0
+  #         "html4Tr"
+  # :           HTML 4.0 Transitional
+  #         "html4Fr"
+  # :           HTML 4.0 with Framesets
+  #         "html5"
+  # :           HTML 5
+  #
+  #
+  #     `:max_multipart_length`
+  # :       Specifies maximum length of multipart data. Can be an Integer scalar
+  #         or a lambda, that will be evaluated when the request is parsed. This
+  #         allows more complex logic to be set when determining whether to accept
+  #         multipart data (e.g. consult a registered users upload allowance)
+  #
+  #         Default is 128 * 1024 * 1024 bytes
+  #
+  #             cgi=CGI.new(:max_multipart_length => 268435456) # simple scalar
+  #
+  #             cgi=CGI.new(:max_multipart_length => -> {check_filesystem}) # lambda
+  #
+  #
+  # `block`
+  # :   If provided, the block is called when an invalid encoding is encountered.
+  #     For example:
+  #
+  #         encoding_errors={}
+  #         cgi=CGI.new(:accept_charset=>"EUC-JP") do |name,value|
+  #           encoding_errors[name] = value
+  #         end
+  #
+  #
+  # Finally, if the CGI object is not created in a standard CGI call environment
+  # (that is, it can't locate REQUEST_METHOD in its environment), then it will run
+  # in "offline" mode.  In this mode, it reads its parameters from the command
+  # line or (failing that) from standard input.  Otherwise, cookies and other
+  # parameters are parsed automatically from the standard CGI locations, which
+  # varies according to the REQUEST_METHOD.
+  #
+  def initialize: (?String tag_maker) ?{ (String name, String value) -> void } -> void
+                | (Hash[Symbol, untyped] options_hash) ?{ (String name, String value) -> void } -> void
+
   # Synonym for $stdin.
   #
   def stdinput: () -> ::IO
 
   # Synonym for $stdout.
   #
   def stdoutput: () -> ::IO
-end
 
-# String for carriage return
-CGI::CR: String
+  CR: String
```

Comment:
I want these constant declarations have comments... Seems like `annotate-with-rdoc` is still broken.

Reply:
I hope https://github.com/ruby/rbs/pull/768 prints the docs.

Reply:
done, it did 👍